modules/spark/src/main/scala/notebook/spark/util/hack/SparkClosureUtilHack.scala

```diff
@@ -0,0 +1,58 @@
+package org.apache.spark.util.notebook.hack
+
+import scala.reflect.runtime.universe
+import org.apache.log4j.AppenderSkeleton
+import org.apache.log4j.spi.LoggingEvent
+
+trait SparkClosureUtilHack {
+  
+  private val CC = org.apache.spark.util.ClosureCleaner
+
+  def appendTo: String => Unit = println
+
+  def clean(o:Object) = {
+    CC.clean(o, java.lang.Boolean.TRUE, java.lang.Boolean.TRUE)
+    // if we reach this point... it's good!
+    "Object cleaned and serializable!"
+  }
+
+  // open the logger from the closure cleaner
+  private val logMethod = CC.getClass.getDeclaredMethod("log")
+  logMethod.setAccessible(true)
+  private val logSlf4J = logMethod.invoke(CC)
+  require(logSlf4J.isInstanceOf[org.slf4j.Logger])
+  private val loggerInF = logSlf4J.getClass.getDeclaredField("logger")
+  loggerInF.setAccessible(true)
+  private val loggerIn = loggerInF.get(logSlf4J)
+  require(loggerIn.isInstanceOf[org.apache.log4j.Logger])
+  private val logger = loggerIn.asInstanceOf[org.apache.log4j.Logger]
```

Comment:
- let's make logger extraction (L19-L28) as a helper function, so it would not leak complexity.
- `CC.log` is `protected` method, so extending `ClosureCleaner` would do the trick. not sure about the methods deeper.